src/docs/generate_oci_documentation.py

```diff
@@ -0,0 +1,358 @@
+                check=True,
+                universal_newlines=True,
+            )
+        except subprocess.CalledProcessError as err:
+            msg = f"Failed to run command: {command!s}"
+            if err.stderr:
+                msg += f" ({err.stderr.strip()!s})"
+            raise subprocess.CalledProcessError(err.returncode, msg)
+
+        return out.stdout.strip()
+
+    def run_skopeo_command(self, cmd: str, args: List[str]) -> Dict:
+        """Builds the Skopeo command and runs it"""
+        command = ["skopeo", cmd]
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            if self.username is not None and self.password is not None:
+                _skopeo_auth_token = base64.b64encode(
+                    f"{self.username}:{self.password}".encode()
+                ).decode()
+                auth_config = {"auths": {self.url: {"auth": _skopeo_auth_token}}}
+                auth_file = os.path.join(tmp_dir, "auth.json")
+                with open(auth_file, "w", encoding="UTF-8") as file:
+                    os.fchmod(file.fileno(), 0o600)
+                    json.dump(auth_config, file)
+                command += ["--authfile", auth_file]
+            command += args
+
+            return json.loads(self.process_run(command))
+
+    def get_arches(self, tag: str, arches_list: List[Dict]) -> List[str]:
+        """
+        Permit to get the arches associated to the tag
+        """
+        logging.info("Getting the arches for %s", tag)
+        arches = []
+        for arch in arches_list:
+            arches.append(arch["platform"]["architecture"])
+        return arches
+
+    def get_lowest_risk(self, tags: List[str]) -> str:
+        """
+        Get the lowest risk associated with the tag
+        """
+        risk_sorted = ["stable", "candidate", "beta", "edge"]
+
+        all_tags_str = " ".join(tags)
+        for risk in risk_sorted:
+            if risk in all_tags_str:
+                return risk
+
+        return "edge"
+
+    def get_tags(self, tag: str, all_tags: List[str]) -> List[str]:
+        """
+        Permit to get the tags associated to the track-base
+        """
+        return [_tag for _tag in all_tags if tag.split("_")[0] == _tag.split("_")[0]]
+
+    def build_releases_data(
+        self,
+        release_lts: Dict,
+        all_tags: List[str],
+    ) -> List[Dict]:
+        """Build the releases info data structure"""
+        releases = []
+        tracks_base = []
+        for tag in all_tags:
+            release_data = {}
+            track = tag.split("-")[0]
+            base = tag.split("_")[0].split("-")[1]
+
+            if tag.split("_")[0] in tracks_base:
+                continue
+
+            tracks_base.append(tag.split("_")[0])
+            release_data["track"] = track
+            release_data["base"] = base
+
+            if release_lts[base]:
+                release_data["type"] = "LTS"
+
+            manifest_list = self.run_skopeo_command(
```

Comment:
this is just because of the arches right? if we have arches in the yaml file, then there's no need to query DH?

Reply:
yes, we only query DH for the arches